### PR TITLE
Add support for --clients to pbench-linpack

### PR DESCRIPTION
**Pull request by ldoktor**
This PR builds on top of: https://github.com/distributed-system-analysis/pbench/pull/2710 and adds the last remaining piece I need to transition from `pbench-run-benchmark` to `pbench-linpack`. Note `pbench-run-benchmark` still allows greater flexibility in linpack configuration, but at this time I don't require that, but it can be added similarly to the threads support.

Note commits 95a5c62..ce0054a are already reviewed and part of the other PR, please focus only on the commit https://github.com/distributed-system-analysis/pbench/pull/2729/commits/d0c6ddbcfa8e4dcf2eb56586f2ce2d3176f7cd1e

### CODE REVIEW

**Review by webbnh — CHANGES_REQUESTED** [review]
Looks good, Lukas.

You need to update `agent/bench-scripts/tests/test-benchmark-clis/test-CL.txt` for your new help text in order to get the tests to pass (and you might want to tweak the help text, first).

I think you're missing the "short option" reference in the `getopts` call.

And, I have a couple of suggestions.

File: agent/bench-scripts/pbench-linpack
Comment (ldoktor) [reply agent/bench-scripts/pbench-linpack:49]: This is pure copy&paste from fio or uperf script ;-) anyway I'll remove that in v2.

File: agent/bench-scripts/pbench-linpack
Comment (ldoktor) [reply agent/bench-scripts/pbench-linpack:58]: ack

File: agent/bench-scripts/pbench-linpack
Comment (ldoktor) [reply agent/bench-scripts/pbench-linpack:217]: ack

File: agent/bench-scripts/pbench-linpack
Comment (ldoktor) [reply agent/bench-scripts/pbench-linpack:229]: ack

File: agent/bench-scripts/postprocess/linpack-prepare-input-file
Comment (ldoktor) [reply agent/bench-scripts/postprocess/linpack-prepare-input-file:229]: True, on the other hand nobody was and is checking the return code, so perhaps I can just remove the handling and leave it to the one implementing it (if needed).

**Comment by ldoktor** [comment]
Tab/space issue, hopefully the selftest will pass this time.

File: agent/bench-scripts/postprocess/linpack-prepare-input-file
Comment (webbnh) [reply agent/bench-scripts/postprocess/linpack-prepare-input-file]: I wouldn't waste any effort on returning a value which is bad, incorrect, or misleading.  So, if you want to return a value here, it should presumably represent the result of the benchmark (or, at least, it should indicate a failure if one occurred in the pipeline); otherwise, I wouldn't bother adding special handling (if, as you say, nothing is looking for it).

**Review by webbnh — DISMISSED** [review]
👍

**Review by portante — CHANGES_REQUESTED** [review]
Needs a rebase now that #2710 has merged.

**Comment by ldoktor** [comment]
Just rebased and used the long variable format in bash.

File: agent/bench-scripts/pbench-linpack
Comment (portante) [inline agent/bench-scripts/pbench-linpack:257]: Based on local unit testing (which I'll post updates for in a PR shortly) this needs to be `cp` since the `linpack.meta` file is only created once per iteration and shared for each sample.

Comment (portante) [reply agent/bench-scripts/pbench-linpack:257]: Both the fix and the unit tests provided in PR https://github.com/ldoktor/pbench/pull/6.

**Review by portante — CHANGES_REQUESTED** [review]

File: agent/bench-scripts/pbench-linpack
Comment (ldoktor) [reply agent/bench-scripts/pbench-linpack:257]: I see, it is actually semi-created during the execution as well (only the `threads` entry) which is why I used `mv`. I'm wondering whether it wouldn't be safer to always create it in `linpack.sh` during the execution, what do you think?

Comment (portante) [reply agent/bench-scripts/pbench-linpack:257]: That PR also fixes the hung test that occurs in the CI.

**Review by webbnh — APPROVED** [review]
I'm not sure what changed, exactly (other than the rebase), but it still looks OK to me.  😀

File: agent/bench-scripts/pbench-linpack
Comment (portante) [reply agent/bench-scripts/pbench-linpack:257]: Once we merge https://github.com/ldoktor/pbench/pull/6 into this PR we should be able to merge this to the `main` branch.

**Comment by portante** [comment]
Superseded by #2746.

**Pull request closed without merging**
